File: backend/src/main/java/com/tasklion/backend/features/taskerService/model/TaskerServiceModel.java
package com.tasklion.backend.features.taskerService.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.tasklion.backend.features.taskerService.serviceCategory.model.ServiceCategoryModel;
import com.tasklion.backend.features.taskerService.serviceReview.model.RatingDetailModel;
import com.tasklion.backend.features.taskerService.serviceReview.model.ServiceReviewModel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Data
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class TaskerServiceModel {

    private String id;
    private String description;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<ServiceReviewModel> reviews;
    private ServiceCategoryModel category;
    private RatingDetailModel ratingDetail;

}

File: frontend/src/app/modules/tasker-service/models/tasker-service.model.ts
import {ServiceReviewModel} from "../service-review/models/service-review.model";
import {ServiceCategoryModel} from "./service-category.model";
import {RatingDetailModel} from "src/app/shared/models/rating-detail.model";

export interface TaskerServiceModel {

  id: number;
  description: string;
  reviews: ServiceReviewModel[];
  category: ServiceCategoryModel;
  ratingDetail: RatingDetailModel;

}
